_posts/2021-01-12-updating-foreman-inventory-with-system-facts.md

```diff
@@ -0,0 +1,323 @@
+
+### Configuring the Salt stack
+
+Foreman Salt plugin also comes with the similar functionality, please take a look at [the manual](https://theforeman.org/plugins/foreman
+_salt/13.0/index.html#2.1.2SaltMasterConfiguration)
+
+## The fruit
+
+Besides that your inventory gets automatically uploaded, there are some other benefits. You notice that OS objects are automatically created and associated with the Hosts. Also Hardware Models, Architectures, Puppet Environments are populated based on what is found in your infrastructure.
+
+If you visit *Hosts -> All Hosts* page, you will see a nice overview. Clicking the host name takes you to the Host detail page. You can see additional information we parsed out of facts, e.g. the IP, MAC, boot time etc. The IP and MAC addresses displayed in the Properties table are addressed by what the fact source considers a primary interface. The NICs tab lists all the network interfaces we received information about.
+
+<p float='left' align='center'>
+  <img src="/static/images/blog_images/2021-01-12-updating-foreman-inventory-with-system-facts/host_details.png" width="700" />
+</p>
+
+Clicking the Facts button in the Details section takes you to the list of all facts for the host. These are the raw values we keep for the host. For example, to find out the python version on your server, try searching for `host = host.example.com and name = ansible_python::version_info`. Similarly, you can search for your kernel version. If you want to see what all python versions you’re running in your infrastructure, click the View Chart button for the specific fact. You can also search for all hosts that run a specific, e.g. a vulnerable version. Click on the chart to filter the hosts represented by the color. The filter field above the table is updated accordingly, so you see an example of how to search for the hosts without using the charts. Searching is only limited by the data you import for your infrastructure.
+
+<p float='left' align='center'>
+  <img src="/static/images/blog_images/2021-01-12-updating-foreman-inventory-with-system-facts/python_version.png" width="700" />
+</p>
+
+The name of the fact differs per fact source, remember, these are the raw values. Some of the facts are processed and unified. You can search hosts by these attributes on the All Hosts page. See the suggestions when you click the filter field. Searching capabilities deserve its own post. For now, please refer to [our manual](https://theforeman.org/manuals/2.3/index.html#4.1.5Searching).
+
+Since you now have the inventory populated, let’s briefly take a look at *Monitor -> Report Templates*. These are the reports you can render from your Foreman. Writing your own template may be covered in some other blog post, however if you’re interested, you can find more information [in our manual](https://theforeman.org/manuals/2.3/index.html#4.11Reports).  
```

Comment:
Could be good to add a section identifier above this as reports are really nice to call out.